be/src/udf/java/java_data_converter.cpp

```diff
@@ -220,6 +223,43 @@ void append_jvalue(MethodTypeDescriptor method_type_desc, Column* col, jvalue va
     }
 }
 
+Status check_type_matched(MethodTypeDescriptor method_type_desc, jobject val) {
+    if (val == nullptr) {
+        return Status::OK();
+    }
+    auto& helper = JVMFunctionHelper::getInstance();
+    auto* env = helper.getEnv();
+
+    switch (method_type_desc.type) {
+#define INSTANCE_OF_TYPE(NAME, TYPE)                          \
+    case NAME: {                                              \
+        if (!env->IsInstanceOf(val, helper.TYPE##_class())) { \
+            return Status::InternalError("Type not matched"); \
```

Comment:
better to give a more detailed message such as what the actual type is and what the expected type is

Reply:
change to `(1064, 'Type not matched, expect class java.lang.Integer, but got class java.lang.String')`